source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -66,7 +67,19 @@ ActiveQuicListener::ActiveQuicListener(Event::Dispatcher& dispatcher,
       crypto_config_.get(), quic_config, &version_manager_, std::move(connection_helper),
       std::move(alarm_factory), quic::kQuicDefaultConnectionIdLength, parent, *config_, stats_,
       per_worker_stats_, dispatcher, listen_socket_);
-  quic_dispatcher_->InitializeWithWriter(new EnvoyQuicPacketWriter(listen_socket_));
+
+  // Create udp_packet_writer
+  udp_packet_writer_ = listener_config.udpPacketWriterFactory()->createUdpPacketWriter(
+      listen_socket_.ioHandle(), listener_config.listenerScope());
+  if (udp_packet_writer_->isBatchMode()) {
+    // UdpPacketWriter* can be downcasted to UdpGsoBatchWriter*, which indirectly inherits
+    // from the quic::QuicPacketWriter class and can be passed to InitializeWithWriter().
+    quic_dispatcher_->InitializeWithWriter(
+        dynamic_cast<Quic::UdpGsoBatchWriter*>(udp_packet_writer_.release()));
+  } else {
+    quic_dispatcher_->InitializeWithWriter(
+        new EnvoyQuicPacketWriter(std::move(udp_packet_writer_)));
+  }
```

Comment:
udpPacketWriter()  will return nullptr after this. I prefer to make udp_packet_writer_ a pointer and always latch to the writer passed around.

Reply:
As discussed offline, having `udp_packet_writer_` now as a raw pointer.